### DISCUSSION TITLE
Support vulnerabilityAlerts for CustomManagers (RegexManagers)

### DISCUSSION THREAD

**Discussion by ryamaguchi0220** [renovatebot/renovate#9622] [Suggest an Idea]
**What would you like Renovate to be able to do?**

Support vulnerabilityAlerts for CustomManagers (RegexManagers).

Assuming that I have a configuration file named `external-libs.json` that manages client-side library dependencies like this

```
{
  "jquery": "https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js",
  "pixi-js": "https://cdn.jsdelivr.net/npm/pixi.js@5.3.8/dist/pixi.min.js",
  "vue-js": "https://cdn.jsdelivr.net/npm/vue@2.6.12/dist/vue.min.js",
  "zxcvbn": "https://cdn.jsdelivr.net/npm/zxcvbn@4.4.2/dist/zxcvbn.js",
  "json-editor-js": "https://cdn.jsdelivr.net/npm/jsoneditor@9.2.0/dist/jsoneditor.min.js",
  "json-editor-css": "https://cdn.jsdelivr.net/npm/jsoneditor@9.2.0/dist/jsoneditor.min.css"
}
```

and `renovate.json` is like this,

```
{
  "vulnerabilityAlerts": {
    "enabled": true
  },
  "regexManagers": [
    {
      "fileMatch": ["external-libs.json"],
      "matchStrings": ["\"https:\/\/cdn.jsdelivr.net\/(?<datasource>.+?)\/(?<depName>.+?)@(?<currentValue>.+?)\/.+\""],
      "versioningTemplate": "semver"
    }
  ]
}
```

This does not work well. In my understanding, this is because Github can not detect vulnerabilities in `external-libs.json`.

**Did you already have any implementation ideas?**

I have no idea. 
In my understanding,  it's impossible as long as `vulnerabilityAlerts` option is based on Github's vulnerability alerts, right?

**Comment by rarkins** [comment]
Yes, not possible because the alerts are based on GitHub's parsing of manifest files

**Reply by ryamaguchi0220** [reply]
I see, thanks for the quick response.
